### PR TITLE
Bind only to localhost by default

### PR DESCRIPTION
**Pull request by platy11**
This PR:
- sets `ev3sim` to only bind to localhost (::1) by default
- adds flags (`--simulator_addr` for `ev3attach` and `--bind_addr` for `ev3sim`) to allow users to listen on all interfaces or use a different port if they want to for whatever reason

Note that this PR targets main, but was based on batched_commands so should be merged after #53. 
I haven't tested this on Windows yet, presumably it works but it would be best to check.

### CODE REVIEW

**Comment by platy11** [comment]
To compare to `batched_commands`: https://github.com/MelbourneHighSchoolRobotics/ev3sim/compare/batched_commands...platy11:bind-to-localhost

File: LICENSE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright 2020 Melbourne High School
+Copyright 2020 the Melbourne High School ev3sim authors
```
Comment (platy11) [inline LICENSE:1]: See https://opensource.stackexchange.com/questions/5508/what-does-and-contributors-in-the-copyright-byline-imply

**Comment by glipR** [comment]
Looks good to me, probably should have done this originally, thanks for the PR :)

File: LICENSE (hunk 1 above)
Comment (glipR) [reply LICENSE:1]: I'd want an ok from @peter-drew or @Dalordish before merging :) I've got no idea if we want to separate ev3sim from Melbourne High School organisation, and move towards "all contributors on the repo"

Comment (Dalordish) [reply LICENSE:1]: At risk of exposing myself as being awake way too late into the night, a few thoughs on this:

**tl;dr this is fine**

Max is correct in that contributors to the repo each own their individual copyright to the code that they write, despite being merged to the repo. This copyright applies regardless of any byline or notice - having it there is just being explicit. 

Fortunately, under the github terms of service, as well as common understanding, although I don't believe this has ever been tested before, contributions to a project containing notice of a license are implicitly licensed under the same terms. Because we use the permissive BSD-3 license, this means we can do whatever we want with the code. (If you want to be explicitly clear, section 6. Contributoins Under Repository License, here https://docs.github.com/en/github/site-policy/github-terms-of-service )

In other words, IANAL, but there shouldn't be any legal difference from including this notice, and we're protected from say, copyright owners revoking our ability to use their contributions by the license.

Comment (Dalordish) [reply LICENSE:1]: I would leave it as "Melbourne High School" rather than "the Melbourne High School". Melbourne High School refers to the legal entity named "Melbourne High School", whereas "the Melbourne High School" is both incorrect in referring to the entitiy (think of saying "The Apple", or "The Google"), as well as in it's other interpretation (the single high school in melbourne)

Comment (Dalordish) [reply LICENSE:1]: Lastly, and this is less relevant to this directly, but attribution raises an important point, which is making sure we recognise contributors for their work - this not only gives people the warm fuzzies, but it also encourages other people to contribute. I think either a  CONTRIBUTORS.md, or using the all-contributors specification (https://github.com/all-contributors/all-contributors) would be nice. Happy to discuss this further in a different thread, but I thought it would be worth noting here in case I forget.

Comment (glipR) [reply LICENSE:1]: all contributors looks great! I might look at adding the bot to our repo later today.

Comment (peter-drew) [reply LICENSE:1]: I like
> Copyright 2020 Melbourne High School and contributors

**Review by peter-drew — CHANGES_REQUESTED** [review]
Either adjust the license as per comments, or move change to separate PR

File: LICENSE (hunk 1 above)
Comment (platy11) [reply LICENSE:1]: Great, I'll change it to "Melbourne High School and contributors" then.
+1 for all-contributors as well.

**Review by Dalordish — APPROVED** [review]
LGTM, ship it 🚀

**Review by glipR — APPROVED** [review]

**Review by peter-drew — APPROVED** [review]
